### PR TITLE
Layout/3

### PR DESCRIPTION
**Pull request by tmarkn**
- added new layout
- split download into dl json and dl zip
- added jump to
- added delete from annotations list
- merged smallImprovements (small bug fixes)
- merged zipUpload (also finished it)
- removed unnecessary files (extra jquery libraries, etc.)

TODO:

- add machine learning connections again
- add model picking (maybe not possible)
- consistent coloring with labels?
- more documentation

### CODE REVIEW

**Review by keithudock — DISMISSED** [review]
Definitely not an engineer design

**Review by ghost — CHANGES_REQUESTED** [review]
Regular search should be case insensitive imo. Also maybe some clarification that its a search bar. I was kinda confused at first.

**Review by ghost — APPROVED** [review]

File: static/js/app.js [hunk 1]
```diff
@@ -76,58 +85,96 @@ $('#sendML').on('click', function () {
...
+    // do each file within zip
+    [].forEach.call(Object.keys(zip.files), function (fileName) {
+      if (tagModel.docIndex(fileName) !== -1) {
+        alert("File already uploaded for: '" + fileName + "' in zip");
+        return;
+      }
+      // mac compressed?
+      if (fileName.match(/^__MACOSX/g) !== null) {
+        alert("Ignored __MACOSX compression file: '" + fileName + "'");
+        return;
```
Comment (tuj52136) [inline static/js/app.js:149]: on macs, zipping files generates a per-file __MACOSX file..  10 zipped txt files will make 10 alert boxes, 100...  It's gonna get real annoying.  Maybe just silently ignore them and log it to the console?

**Review by tuj52136 — CHANGES_REQUESTED** [review]
Looks great. Just the runaway alerts when ignoring mac compression files

File: static/js/app.js [hunk 2]
```diff
@@ -560,10 +590,10 @@ function makeRandColor() {
   });
 }
 
-//
-function loadJsonData(data, obliterate = false, filename = "") {
-  console.log('Displaying new data from mlalgorithm');
+// import json data
+function loadJsonData(data, filename = "", obliterate = false, ) {
```
Comment (tuj52136) [inline static/js/app.js:594]: I don't think we use filename here, unless it's something we're working on

File: static/js/app.js [hunk 3]
```diff
@@ -33,10 +33,19 @@ $('#download').on('click', function () {
...
+  if (tagModel.openDocs.length === 0) {
+    alert('Error: No data to download!');
+    return;
+  }
+  var blob = new Blob([tagModel.exportAsString()], { type: 'application/JSON' });
+  saveAs(blob, "annotations.json");
+});
 
 // send to mldata
 $('#sendML').on('click', function () {
```
Comment (tuj52136) [inline static/js/app.js:51]: this button id changed

File: static/js/app.js (hunk 2 above)
Comment (tmarkn) [reply static/js/app.js:594]: The idea was to use it for alerts

File: static/js/app.js (hunk 1 above)
Comment (tmarkn) [reply static/js/app.js:149]: Well, right now, it's a problem with both zip and json files.

Comment (tuj52136) [reply static/js/app.js:149]: actually yeah i see it now. Then I'll approve this and we can fix that on its own

**Review by tuj52136 — APPROVED** [review]

**Pull request merged**